app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -7,16 +7,25 @@ module Foreman::Controller::SmartProxyAuth
   module ClassMethods
     def add_smart_proxy_filters(actions, options = {})
       skip_before_filter :require_login, :only => actions
-      skip_before_filter :require_ssl, :only => actions
       skip_before_filter :authorize, :only => actions
       skip_before_filter :verify_authenticity_token, :only => actions
       skip_before_filter :set_taxonomy, :only => actions
       skip_before_filter :session_expiry, :update_activity_time, :only => actions
       before_filter(:only => actions) { require_smart_proxy_or_login(options[:features]) }
       attr_reader :detected_proxy
+      define_method("require_ssl_with_dont?") do
+        send("require_ssl?") unless actions
+      end
+      alias_method_chain :require_ssl?, :dont
     end
   end
 
+  protected
+
+  def require_ssl_with_dont?
```

Comment:
is this required?  I can't see how it'd be called, since you define the same method in the add_smart_proxy_filters method, and only call alias_method_chain at the same time.